apps/publisher/themes/default/js/taxonomy-view-asset.js

```diff
@@ -0,0 +1,193 @@
+        },
+        error: function () {
+        }
+    });
+
+
+    function unique(list) {
+        var result = [];
+        $.each(list, function(i, e) {
+            if ($.inArray(e, result) == -1) result.push(e);
+        });
+        return result;
+    }
+
+    var currentNode;
+    var jsTreeView = $('#jstree-taxonomy');
+
+    jsTreeView.jstree({
+        conditionalselect : function (node, event) {
+            var tree = $('#jstree-taxonomy').jstree(true);
+            if (tree.is_leaf(node) || event.toElement.className == "jstree-icon jstree-checkbox" || event.toElement.className == "jstree-icon jstree-checkbox jstree-undetermined") {
+                if (!tree.is_leaf(node)) {
+                    tree.open_all(node);
+                }
+                return true;
+
+            } else {
+                //tree.open_node(node);
+                if (tree.is_open(node)) {
+                    tree.close_node(node);
+                }else {
+                    tree.open_node(node);
+                }
+                return false;
+            }
+
+        },
+        plugins: ["checkbox", "conditionalselect"],
+        core: {
+            data: {
+                animation: 0,
+                check_callback: true,
+                url: function (node) {
+                    // modify url when clicking on tree nodes to load children
+                    if (node.id === '#') {
+                        return caramel.context + '/apis/taxonomies';
+                    } else {
+                        return caramel.context + '/apis/taxonomies?terms=' + node.id + '/children';
+                    }
+                },
+                data: function (node) {
+                    currentNode = node;
+                },
+                success: function (data) {
+
+                    var children;
+                    try {
+                        children = Array.isArray(data[0].children);
+                    } catch (e) {
+
+                    }
+
+                    if (children) {
+                        data[0].id = data[0].elementName;
+                        if (data[0].text == "") {
+                            data[0].text = data[0].elementName;
+                        }
+                        for (var i = 0; i < data[0].children.length; i++) {
+                            data[0].children[i].id = data[0].elementName + "/" + data[0].children[i].elementName;
+                            if (data[0].children[i].text == "") {
+                                data[0].children[i].text = data[0].children[i].elementName;
+                            }
+                        }
+                    } else {
+                        for (var i = 0; i < (data.length); i++) {
+                            data[i].id = currentNode.original.id + "/" + data[i].elementName;
+                            if (data[i].text == "") {
+                                data[i].text = data[i].elementName;
+                            }
+
```

Comment:
To make this code more concise it can be written as follows:
- Reference the 0th element only once
- Use JS Array methods to more naturally express iteration

var dataAtZeroTh = data[0] || { text: '', children:[] } ;
dataAtZeroTh.text = (dataAtZeroTh.text === "") ? dataAtZeroTh.elementName;
dataAtZeroTh.children.forEach(function(dataItem,index) {
   //You logic
});